### PR TITLE
Simple CMake in AzureDevOps

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -326,5 +326,14 @@ ASALocalRun/
...
 .mfractor/
+
+# CMake
+CMakeCache.txt
+CMakeFiles/
+
+*.vcxproj
+*.vcxproj.filters
+*.sln
+cmake_install.cmake
```
Comment (joshfree) [inline .gitignore]: missing new line

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+add_subdirectory(sdk/keyvault/keyvault)
```
Comment (joshfree) [inline CMakeLists.txt]: missing new line

File: sdk/keyvault/keyvault/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+cmake_minimum_required (VERSION 3.0)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/keyvault/CMakeLists.txt]: This minimum version doesn't match what's in the template https://github.com/Azure/azure-sdk-for-c/pull/2/files (3.15)

Comment (joshfree) [inline sdk/keyvault/keyvault/CMakeLists.txt]: any reason not to just use 3.15.1 since this is a brand new repo?

Comment (sergey-shandar) [reply sdk/keyvault/keyvault/CMakeLists.txt]: Yes, I started it before. I will update to the latest template as soon as it merged.

Comment (sergey-shandar) [reply sdk/keyvault/keyvault/CMakeLists.txt]: No, I'm just trying to integrate CMake with AzureDevOps. I will update to the latest template as soon as it merged to master

Comment (sergey-shandar) [reply sdk/keyvault/keyvault/CMakeLists.txt]: BTW, it looks like, AzureDevOps Ubuntu 16 has a CMake version 3.12. We will need either to use 3.12 or install 3.15 with some kind of `apt-get`.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      Linux:
+        vm.image: 'ubuntu-16.04'
+      Win:
+        vm.image: 'windows-2019'
+      MacOS:
+        vm.image: 'macOS-10.14'
+  pool:
+    vmImage: $(vm.image)
+  steps:
+  - task: CMake@1
```
Comment (sergey-shandar) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: @bsiegel currently, it tries to build all projects by using `CMakeLists.txt` in the root. I will need to update it to is `$(ServiceDirectory)`.

**Comment by sergey-shandar** [comment]
@jhendrixMSFT, @barcharcraz please, review :-)

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+cmake_minimum_required (VERSION 3.12)
+project(az)
+add_subdirectory(sdk/storage/storage)
```
Comment (jhendrixMSFT) [inline CMakeLists.txt]: Why is /storage repeated, or is this just temporary to give the build system something to build and will be removed later?

File: build.ps1 [hunk 6]
```diff
@@ -0,0 +1,4 @@
+$location = $PSScriptRoot
+$build = Join-Path $location "build"
+CMake -S $location -B $build
+CMake --build $build
```
Comment (jhendrixMSFT) [inline build.ps1]: Missing new-line.

File: sdk/storage/storage/build.ps1 [hunk 7]
```diff
@@ -0,0 +1,4 @@
+$location = $PSScriptRoot
+$build = Join-Path $location "build"
+CMake -S $location -B $build
+CMake --build $build
```
Comment (jhendrixMSFT) [inline sdk/storage/storage/build.ps1]: Missing new-line.

File: build.ps1 (hunk 6 above)
Comment (jhendrixMSFT) [reply build.ps1]: Also does this mean we're standardizing on PS for cross-plat build scripting?

Comment (sergey-shandar) [reply build.ps1]: @jhendrixMSFT, @bsiegel  as far as I know, PS core is the only one that is installed by default on all Azure Dev Ops hosts (Windows, Linux, Mac OS X). This specific `build.ps1` is not used by CI and it's for developers. The problem is that by default CMake will output everything in current directory :-)

File: CMakeLists.txt (hunk 5 above)
Comment (sergey-shandar) [reply CMakeLists.txt]: I'm following a pattern from other SDKs. AFAIK, some services may have several libraries/projects etc.

File: build.ps1 (hunk 6 above)
Comment (barcharcraz) [reply build.ps1]: IMO it's better to just not add these kinds of build scripts, but that's just me.

File: sdk/keyvault/keyvault/CMakeLists.txt (hunk 3 above)
Comment (barcharcraz) [reply sdk/keyvault/keyvault/CMakeLists.txt]: this should be at least 3.12, no reason to revert to a lower version here after setting 3.12 in the parent project

**Pull request merged**